cloudinit/config/schema.py

```diff
@@ -1140,6 +1154,12 @@ def validate_cloudconfig_file(
                 network_config=cloudconfig, strict=True, annotate=annotate
```

Comment:
As the code stands, `validate_cloudconfig_file` partially reimplements what `validate_cloudconfig_schema` does, with no apparent reason but printing message instead of logging them (because `validate_cloudconfig_file` is only called to handle the `cloud-init schema` command). Unless I am missing another reason, and if so it should be documented, why don't we simplify this like:

```diff
diff --git a/cloudinit/config/schema.py b/cloudinit/config/schema.py
index 407cade0d..7468fb27b 100644
--- a/cloudinit/config/schema.py
+++ b/cloudinit/config/schema.py
@@ -1150,16 +1150,6 @@ def validate_cloudconfig_file(
         network_version = network_schema_version(cloudconfig)
         if network_version == 2:
             schema_type = SchemaType.NETWORK_CONFIG_V2
-            if netplan_validate_network_schema(
-                network_config=cloudconfig, strict=True, annotate=annotate
-            ):
-                return True  # schema validation performed by netplan
-            elif netplan_available():
-                print(
-                    "Skipping network-config schema validation for version: 2."
-                    " No netplan API available."
-                )
-                return False
         elif network_version == 1:
             schema_type = SchemaType.NETWORK_CONFIG_V1
             # refresh schema since NETWORK_CONFIG defaults to V2
```

and handle the printing-message-problem by returning an enum in `validate_cloudconfig_schema`?

Reply:
Either returning an enum from validate_cloudconfig_schema, or maybe raising specialized errors from validate_cloudconfig_schema which would allow us to differentiate no JSON schema importable from no netplan API importable. I think this warrants inclusion in a general refactor of these two functions that we should carry separate from this PR.